filesystems/watcher.go

```diff
@@ -701,10 +738,28 @@ func buildFileListInDir2(dirName string, fileList []string, checkModifiedTimeThr
 				continue
 			}
 
+			elem, ok := dedupUpload[fileInfo.Name()]
```

Comment:
I think that it'd be simpler to use a cache for the dedupe upload storage. We use github.com/karlseguin/ccache/v2 elsewhere in the code.

You can have the key be something like:
{Filename, Ctime}, and the value be {Size, Mtime}
And if size, mtime changes, or the lookup in cache is a miss, then upload.

Reply:
Also, IMHO, the dedupe logic should live in watcher.uploadLogfile.

Reply:
I'm using ccache now but chose to omit ctime because it requires separate system calls to get the ctime and didn't see a benefit over using size and mtime (could be convinced otherwise).

Reply:
Also moved the dedupe logic to uploadLogFile.